### ISSUE TITLE
eslint-recommended does not raise error when indent-rule default are violated

### ISSUE THREAD

**Issue by GradedJestRisk** [eslint/eslint#13703]
Bare-metal bug reproduction can be found in the repository  [here](https://github.com/GradedJestRisk/eslint-indent)
No open issues matching

**Tell us about your environment**

Linux, all npm latest versions, no globals
* **ESLint Version:** v7.9.0  (no global)
* **Node Version:**  v14.11.0
* **npm Version:** v6.14.8

Reproduced on another machine (can providedetail if needed)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
No parser used

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "node": true,
        "es2021": true
    },
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "extends": [
        'eslint:recommended'
    ],
    "rules": {
    }
};
```
</details>

**What did you do?**

```js
let  a, b, c, e, f ;
if (a) {
  // eslint-disable-next-line no-unused-vars
  b=c;
  // eslint-disable-next-line no-inner-declarations,no-unused-vars
  function foo(d) {
    // eslint-disable-next-line no-unused-vars
    e=f;
  }
}
```

```bash
npx eslint foo.js
```

**What did you expect to happen?**
The following message 
`$ 11:1 error  Expected indentation of 4 spaces but found 2  indent`

Full rationale in
[the sample code](https://github.com/GradedJestRisk/eslint-indent/blob/master/foo.js)

**What actually happened? Please include the actual, raw output from ESLint.**
No message (return code 0)

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by mdjermanovic** [comment]
Hi @GradedJestRisk, thanks for the issue!

`eslint:recommended` enables only rules that report common problems, mostly possible errors in the code.

`indent` is not among them, since it's a stylistic rule.

You can find which rules are enabled by `eslint:recommended` on our [rules page](https://eslint.org/docs/rules/). Those rules have a check mark on the left side.

If you want to enable all builtin rules, you can use `eslint:all` configuration.

**Comment by GradedJestRisk** [comment]
Thank you @mdjermanovic
I don't know why I didn't come up with it, this is clearly stated in rules page. 
> No rules are enabled by default. The "extends": "eslint:recommended" property in a configuration file enables rules that report common problems, which have a check mark below.

Sorry for the unconvenience !
